hi-ml-azure/src/health_azure/himl.py

```diff
@@ -921,7 +923,8 @@ def submit_to_azure_if_needed(  # type: ignore
             )
             if entry_script is None:
                 entry_script = Path(sys.argv[0])
-            script_params = script_params or sys.argv[1:]
+
+            script_params = script_params or [param for param in sys.argv[1:] if param != AZUREML_COMMANDLINE_FLAG]
```

Comment:
you will also need a matching change for submission using the V1 SDK. Consequently, it might make sense to move that messing with `script_params` to before line 883, so that both codepaths are affected in the same way. Please also add tests.